src/xdShare.sol

```diff
@@ -115,9 +116,37 @@ contract xdShare is IxdShare, Ownable, ERC4626, ReentrancyGuard {
         underlyingDShare = newUnderlyingDShare;
     }
 
-    /// @notice Converts all parent dshare vault balances to the current dShare token.
-    // TODO: call tokenmanager sweepconvert
-    // function sweepConvert() external onlyOwner {}
+    /**
+     * @dev Convert all pre-split tokens held by the vault to the
+     * current version of the dShare token. It iteratively approves the TokenManager to handle
+     * the conversion of each parent token found, updating the references to the current and parent tokens
+     * at each iteration. Once all parent tokens have been processed, it triggers the conversion process
+     * in the TokenManager by calling its `sweepConvert` function.
+     */
+    function sweepConvert() external onlyOwner {
```

Comment:
After seeing how much of a mess this creates with approvals, rather than iterating over all the historical tokens, sweepConvert should take a token as input. If that token is a historical pre-split token, then convert it to the current. That way this method can manage a single approval and can be called once per token as needed.